src/magnet_link_parse_error.rs

```diff
@@ -0,0 +1,21 @@
+use crate::common::*;
+
+#[derive(Debug, Snafu)]
+#[snafu(visibility(pub(crate)))]
+pub(crate) enum MagnetLinkParseError {
```

Comment:
Some of these messages will be redundant when printed in context, i.e. with `Failed to parse magnet link:` from the outer `Error` message.

Reply:
Tests should be added that use malformed magnet links to trigger all of these error variants, and check that their contents are correct.